### DISCUSSION TITLE
关于minmax中目标函数的类型限制

### DISCUSSION THREAD

**Discussion by Junjie-Wang-scut** [XiongPengNUS/rsome#57] [Q&A]
作者您好，我是最近刚刚尝试使用RSOME和鲁棒优化的用户，请问RSOME的minmax函数中对目标函数的类型是否存在限制，我在minmax函数中使用abs函数作为目标函数时会提示ypeError: bad operand type for abs(): 'RoAffine'，请问造成这个报错的原因是什么呢，感谢。

**Comment by XiongPengNUS** [comment]
在RSOME中，所有的凸函数，包括`abs()`，都不能用在含有随机变量的表达式上。比如对于决策变量`x`和随机变量`z`，`abs(x)`是支持的，但是`abs(x*z)`不支持。如果要最小化`x*z`的绝对值，可以引入一个辅助变量`y`，写成：

```python
model.min(y)
model.st(x*z <= y, x*z >= -y)
```
